spec/03-plugins/26-oauth2/03-access_spec.lua

```diff
@@ -1083,6 +1083,24 @@ describe("Plugin: oauth2 (access)", function()
         local body = assert.res_status(200, res)
         assert.is_table(ngx.re.match(body, [[^\{"token_type":"bearer","access_token":"[\w]{32,32}","expires_in":5\}$]]))
       end)
+      it("returns success with authorization header and client_id body param", function()
+        local res = assert(proxy_ssl_client:send {
+          method = "POST",
+          path = "/oauth2/token",
+          body = {
+            client_id = "clientid123",
+            scope = "email",
+            grant_type = "client_credentials"
+          },
+          headers = {
+            ["Host"] = "oauth2_4.com",
+            ["Content-Type"] = "application/json",
+            Authorization = "Basic Y2xpZW50aWQxMjM6c2VjcmV0MTIz"
+          }
+        })
+        local body = assert.res_status(200, res)
+        assert.is_table(ngx.re.match(body, [[^\{"token_type":"bearer","access_token":"[\w]{32,32}","expires_in":5\}$]]))
```

Comment:
oof, these tests are in a really poor form. we should be decoding the returned JSON body and examining each element we care about, not presuming some details about the form of the encoded payload. this isnt a blocker for this review, since it looks like a copypasta from previous commits, more of a note that we should clean up this test in the future :)